### PR TITLE
Create post_advance_work for TurbulenceModel

### PR DESCRIPTION
**Pull request by gantech**
Also create a specific instance of post_advance_work for KsgsM84 to populate an sdr field to work with IDDES model in Nalu-wind

### CODE REVIEW

File: amr-wind/turbulence/LES/OneEqKsgs.cpp [hunk 1]
```diff
@@ -206,6 +210,47 @@ void OneEqKsgsM84<Transport>::update_scalar_diff(
...
 }
 
+template <typename Transport>
+void OneEqKsgsM84<Transport>::post_advance_work() {
+
+    BL_PROFILE("amr-wind::" + this->identifier() + "::post_advance_work");
+
+    if ( m_hybrid_rl )   {
+        // Update sdr field based on sfs ke
+
```
Comment (sayerhs) [inline amr-wind/turbulence/LES/OneEqKsgs.cpp:220]:
```suggestion
    if (!m_hybrid_rl) return; 
    
    BL_PROFILE("amr-wind::" + this->identifier() + "::post_advance_work");
```

File: amr-wind/turbulence/LES/OneEqKsgs.cpp [hunk 2]
```diff
@@ -206,6 +210,47 @@ void OneEqKsgsM84<Transport>::update_scalar_diff(
...
+        for (int lev = 0; lev < nlevels; ++lev) {
+            const auto& geom = geom_vec[lev];
+
+            const amrex::Real dx = geom.CellSize()[0];
+            const amrex::Real dy = geom.CellSize()[1];
+            const amrex::Real dz = geom.CellSize()[2];
+            const amrex::Real ds = std::cbrt(dx * dy * dz);
+
+            for (amrex::MFIter mfi(tke(lev)); mfi.isValid(); ++mfi) {
+                const auto& bx = mfi.tilebox();
```
Comment (sayerhs) [inline amr-wind/turbulence/LES/OneEqKsgs.cpp:237]: Use `growntilebox` here to eliminate `fillpatch` communication.

File: amr-wind/turbulence/LES/OneEqKsgs.cpp [hunk 3]
```diff
@@ -206,6 +210,47 @@ void OneEqKsgsM84<Transport>::update_scalar_diff(
...
+
+                amrex::ParallelFor(
+                    bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+                        sdr_arr(i, j, k) = std::sqrt(tke_arr(i,j,k))/(Ce * ds);
+                    });
+            }
+        }
+
+        sdr.fillpatch(this->m_sim.time().current_time());
+
```
Comment (sayerhs) [inline amr-wind/turbulence/LES/OneEqKsgs.cpp:249]:
```suggestion
```

File: amr-wind/turbulence/LES/OneEqKsgs.cpp [hunk 4]
```diff
@@ -48,8 +48,12 @@ OneEqKsgsM84<Transport>::OneEqKsgsM84(CFDSim& sim)
     {
         amrex::ParmParse pp("ABL");
         pp.get("reference_temperature", m_ref_theta);
+        pp.query("enable_hybrid_rl_mode", m_hybrid_rl);
     }
 
+    if (m_hybrid_rl)
+        m_sdr = &(sim.repo().declare_field("sdr", 1, 1, 1));
```
Comment (sayerhs) [inline amr-wind/turbulence/LES/OneEqKsgs.cpp:55]: Overset connectivity with TIOGA currently requires the _same number of ghost cells_ for all fields being exchanged. Use the same number of ghost cells as TKE field here. 

```C++
this->m_sdr = &(sim.repo().declare_field("sdr", 1, this->m_tke.num_grow()[0], 1);
```

Also if we decide to keep `fillpatch` we should also set default fillpatch operator 

```C++
this->m_sdr->set_default_fill_patch_op(this->m_sim.time());
```

File: amr-wind/turbulence/LES/OneEqKsgs.H [hunk 5]
```diff
@@ -31,6 +31,7 @@ protected:
     //! Turbulence constant
     amrex::Real m_Ce{0.1};
     amrex::Real m_Ceps{0.93};
+
```
Comment (sayerhs) [inline amr-wind/turbulence/LES/OneEqKsgs.H:34]:
```suggestion
```

File: amr-wind/turbulence/LES/OneEqKsgs.cpp (hunk 1 above)
Comment (gantech) [reply amr-wind/turbulence/LES/OneEqKsgs.cpp]: Fixed in 8ecbe368e5e6815e45ca2be481488572dfd0ea7f

File: amr-wind/turbulence/LES/OneEqKsgs.cpp (hunk 3 above)
Comment (gantech) [reply amr-wind/turbulence/LES/OneEqKsgs.cpp]: Fixed in 8ecbe368e5e6815e45ca2be481488572dfd0ea7f

File: amr-wind/turbulence/LES/OneEqKsgs.H (hunk 5 above)
Comment (gantech) [reply amr-wind/turbulence/LES/OneEqKsgs.H]: Fixed in 8ecbe368e5e6815e45ca2be481488572dfd0ea7f

File: amr-wind/turbulence/LES/OneEqKsgs.cpp (hunk 4 above)
Comment (gantech) [reply amr-wind/turbulence/LES/OneEqKsgs.cpp]: Fixed in 8ecbe368e5e6815e45ca2be481488572dfd0ea7f

File: amr-wind/turbulence/LES/OneEqKsgs.cpp (hunk 2 above)
Comment (gantech) [reply amr-wind/turbulence/LES/OneEqKsgs.cpp]: Fixed in 8ecbe368e5e6815e45ca2be481488572dfd0ea7f

**Pull request closed without merging**